.github/workflows/dotnet-sample-app-s3-deploy.yml

```
@@ -42,13 +46,55 @@ jobs:
           role-to-assume: arn:aws:iam::${{ env.ACCOUNT_ID }}:role/${{ secrets.E2E_TEST_ROLE_ARN }}
           aws-region: ${{ matrix.aws-region }}
 
+      - name: Update .NET version if not default
+        working-directory: sample-apps/dotnet
+        run: |
+          if [ "${DOTNET_VERSION}" != "${DEFAULT_VERSION}" ]; then
+            echo "Updating .NET version from ${DEFAULT_VERSION} to ${DOTNET_VERSION}"
+            
+            # Function to update version in files
+            update_version() {
+              local file=$1
+              if [ ! -f "$file" ]; then
+                echo "Error: File $file not found"
+                exit 1
+              fi
+              
+              # Update TargetFramework in csproj files
+              if [[ $file == *.csproj ]]; then
+                sed -i "s/<TargetFramework>netcoreapp${DEFAULT_VERSION}<\/TargetFramework>/<TargetFramework>netcoreapp${DOTNET_VERSION}<\/TargetFramework>/g" "$file"
```

A large portion of the string is redundant. Just look for ${DOTNET_VERSION}

```suggestion
                sed -i "s/${DEFAULT_VERSION}/${DOTNET_VERSION}/g" "$file"
```

---

Hey Harry, currently what you suggested works but in the future
This command would replace ALL instances of the version number anywhere in the file, which could be problematic because:
It would replace the version number even if it appears in comments
It would replace the version number in any other XML elements or attributes
It might replace version numbers that shouldn't be changed
It could accidentally replace partial matches (for example, if DEFAULT_VERSION is "6.0" and the file contains "16.0" somewhere), so is it okay to not change this?